File: 7. Reverse Integer.cpp
class Solution {
public:
    int reverse(int x) {
        bool flag = x < 0;
        int res;        
        if (flag)
            x = -x;
        if (x < 0)
            return 0;              
        res = 0;
        while(x > 0){       
            if (x < 10) {
               if ((INT_MAX - x) * 1.0 / 10 < res)
                    return 0;
            }
            res = res * 10 + x % 10;   
            x /= 10;
        }
        if (flag)
            res = -res;        
        return res;                    
    }
};

File: 5-Problem-A-Day/7. Reverse Integer.java
class Solution {
    public int reverse(int x) {
        int ans = 0;
        while (x != 0) {
            int tmp = x % 10;                        
            if ((ans > 0 && ans > (Integer.MAX_VALUE - tmp) / 10) || (ans < 0 && ans < (Integer.MIN_VALUE - tmp) / 10))
                return 0;
            ans = ans * 10 + tmp;
            x /= 10;
        }
        return ans;
    }
}
